civil.go

```diff
@@ -415,6 +423,25 @@ func (dt *DateTime) UnmarshalText(data []byte) error {
 
 // UnmarshalJSON implements encoding/json Unmarshaler interface
 func (dt *DateTime) UnmarshalJSON(data []byte) error {
+	if dt == nil {
+		return errors.New("nil receiver")
+	}
+	if tIdx := bytes.IndexAny(data, "Tt"); tIdx > 0 {
```

Comment:
We should eliminate this `if` and the original code below it.

If `T` isn't found, it can't be a valid `DateTime`

Reply:
My rework of this should also address the above issue to some extent.